src/crate/client/sqlalchemy/monkey.py

```diff
@@ -0,0 +1,44 @@
+from crate.client.sqlalchemy.sa_version import SA_VERSION, SA_1_4
+
+
+def sqlalchemy_without_cresultproxy():
+    """
+    When running on SQLAlchemy 1.4 with the overhauled ``cresultproxy``
+    C-implementation, UPDATE statements will raise::
+
+        RuntimeError: number of values in row (3) differ from number of column processors (2)
+
+    The problem can be reproduced by running::
+
+        ./bin/test --test test_assign_to_craty_type_after_commit
+
+    This function is a workaround to purge the ``cresultproxy``
+    extension at runtime by using monkeypatching.
+
+    The reason might be the CrateDB-specific amendments within
+    ``visit_update_14`` or even general problems with the new
+    architecture of SQLAlchemy 1.4 vs. the needs of CrateDB.
+
+    See also:
+    - https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#rowproxy-is-no-longer-a-proxy-is-now-called-row-and-behaves-like-an-enhanced-named-tuple
+    - https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_0b1/lib/sqlalchemy/cextension/resultproxy.c
+    """
```

Comment:
The monkeypatch was removed with ed2e1e14. It was needed because there were inaccuracies when setting up fake cursors within the test cases. SQLAlchemy <1.4 might have been more forgiving on that, but >=1.4 isn't.